### PR TITLE
Zero-copy read for 0-compression archives

### PR DESCRIPTION
**Pull request by revonateB0T**

### CODE REVIEW

File: app/src/main/cpp/archive.c [hunk 1]
```diff
@@ -387,15 +387,31 @@ Java_com_hippo_ehviewer_jni_ArchiveKt_extractToByteBuffer(JNIEnv *env, jclass th
...
+        if (zero_copy) {
+            ctx->using = 0;
+            return (*env)->NewDirectByteBuffer(env, inner_buff, inner_buff_len);
+        } else {
+            size_t bytes = 0;
+read:
+            memcpy(addr + output_ofs, inner_buff, inner_buff_len);
+            bytes += inner_buff_len;
+            ret = archive_read_data_block(ctx->arc, (const void **) &inner_buff, &inner_buff_len, &output_ofs);
+            if (inner_buff_len && !ret) goto read;
```
Comment (FooIbar) [inline app/src/main/cpp/archive.c:404]: Reason for using goto instead of loop?

**Comment by revonateB0T** [comment]
Btw I don't think we need to avoid the `mmap` for anon pages pool
There is two reasons:
1: Maybe not all entries can do zero-copy read, we still need a large buffer.
2: Unless we touch the anon-pages pool address-region, physical pages will not be populated in kernel, no extra memory pressure, just some `vma` allocation

File: app/src/main/cpp/archive.c (hunk 1 above)
Comment (revonateB0T) [reply app/src/main/cpp/archive.c:404]: I don't think this `goto` is harmful.

Comment (revonateB0T) [reply app/src/main/cpp/archive.c]: Done

**Pull request merged**